presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java

```diff
@@ -159,5 +159,6 @@ public void testCreateTableAsSelect()
                 .executeQuery(sql);
 
         assertThat(queryResult).matches(PRESTO_NATION_RESULT);
+        onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", CREATE_TABLE_AS_SELECT));
```

Comment:
This won't be triggered if previous queries fail. You need some sort of final-block but without hiding the reason for an actual failure. Have a look how we have this done in `AbstractTestHiveClient` (method `dropTable`). Or, similar approach in `AlterTableTests`, with `dropTestTables` method.